tests/api/ocsp.c

```diff
@@ -0,0 +1,552 @@
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+#if !defined(WOLFSSL_USER_SETTINGS) && !defined(WOLFSSL_NO_OPTIONS_H)
+    #include <wolfssl/options.h>
+#endif
+#include <wolfssl/wolfcrypt/settings.h>
+
+#include <tests/api/ocsp.h>
+#include <tests/api/ocsp_test_blobs.h>
+#include <tests/unit.h>
+#include <wolfssl/internal.h>
+#include <wolfssl/ocsp.h>
+#include <wolfssl/ssl.h>
+
+#if defined(HAVE_OCSP)
+struct ocsp_cb_ctx {
+    byte* response;
+    int responseSz;
+};
+
+struct test_conf {
+    unsigned char* resp;
+    int respSz;
+    unsigned char* ca0;
+    int ca0Sz;
+    unsigned char* ca1;
+    int ca1Sz;
+    unsigned char* targetCert;
+    int targetCertSz;
+};
+
+static int ocsp_cb(void* ctx, const char* url, int urlSz, unsigned char* req,
+    int reqSz, unsigned char** respBuf)
+{
+    struct ocsp_cb_ctx* cb_ctx = (struct ocsp_cb_ctx*)ctx;
+    (void)url;
+    (void)urlSz;
+    (void)req;
+    (void)reqSz;
+
+    *respBuf = cb_ctx->response;
+    return cb_ctx->responseSz;
+}
+
+static int test_ocsp_response_with_cm(struct test_conf* c, int expectedRet)
+{
+    EXPECT_DECLS;
+    WOLFSSL_CERT_MANAGER* cm = NULL;
+    struct ocsp_cb_ctx cb_ctx;
+    int ret;
+
+    cm = wolfSSL_CertManagerNew();
+    ExpectNotNull(cm);
+    ret = wolfSSL_CertManagerEnableOCSP(cm,
+        WOLFSSL_OCSP_URL_OVERRIDE | WOLFSSL_OCSP_NO_NONCE);
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
```

Comment:
The benefit of doing everything inside the `Expect` API is that once one instruction fails, the next ones are skipped. This will make the output logs easier to parse as the last logs are the ones that failed. By doing it outside of `Expect` the logs will show a bunch of other things being done after the initial failure.
```suggestion
    ExpectIntEQ(wolfSSL_CertManagerEnableOCSP(cm,
        WOLFSSL_OCSP_URL_OVERRIDE | WOLFSSL_OCSP_NO_NONCE), WOLFSSL_SUCCESS);
```